### ISSUE TITLE
Dispatch - Some phrase lists aren't being imported during dispatch creation

### ISSUE THREAD

**Issue by NC2408** [microsoft/botbuilder-tools#1322]
## Tool
Name: Dispatch
Version: 1.5.5
OS: Windows Server 2019 Datacenter - Version 10.0.17763

## Describe the bug
We have two LUIS models that each have distinct phrase lists (model 1 has 3 phrase lists, model 2 has 6) and a QnA KB that we want to use in a dispatch model. When running this dispatch creation process, 3 of the phrase lists from model 2 do not get imported as part of that process. It is consistently the same 3 phrase lists that are missing unless I exclude model 1 and delete the other 3 phrase lists from model 2 prior to creating the dispatch model. The longest phrase list we have only has 76 entries so we are well under the limit. It appears as if it times out and moves on to the next step of the process before it is able to go through each of the phrase lists

[bug]

**Comment by dmvtech** [comment]
I'm going to attempt to reproduce this. I'll let you know if I have any questions. Thanks @NC2408

**Comment by dmvtech** [comment]
I'm able to reproduce similarly. 

I have:
model1: 3 phrase lists, all interchangeable
model2: 6 phrase lists, 3 interchangeable, 3 not interchangeable

In my dispatch model I have 7 (as opposed to your 6). 2 of my interchangeable ones are missing.

I don't think it has to do with the interchangeable property, but would you let me know what you have set for your 9 phrase lists? (just a total. interchangeable/not)

I will continue to investigate.

**Comment by hcyang** [comment]
Hi,

During creating a dispatch LUIS model, Dispatch CLI will try to merge phrase lists from sub apps in order to avoid hitting LUIS's 10-phrase-lists limit. In the scenario, Dispatch will merge the 3 non-interchangeable phrase lists into 1 (before splitting it again if its size is over LUIS' phrase list size limit). The interchangeable phrases might get merged if they share the same name.

Please share with me your two LUIS apps and I can take a look if there is a bug in the merge logic.
In the meantime, please check if any of the phrase list items are missing from the merged lists.

Hung-chih

**Comment by tsuwandy** [comment]
@NC2408, if possible could you share your exported LUIS files to hunyang@microsoft.com so he could take a look at the issue?  Thanks!

**Comment by hcyang** [comment]
Dana shared 2 LUIS apps and their Dispatch app for repro. As Dana indicated, there are, in total, 9 phrase lists from two the sub models, 6 of them are interchangeable, and 3 non-interchangeable. The 3 non-interchangeable lists were merged into one called “ni_0” in the dispatch model. The other 6 are copied over and processed into “i_0” to “i_5.” Thus the 9 original phrase lists have become 7, as expected.

Weapons -> i_0
Armor -> i_1
Biome, ClothingItems, Emotions -> ni_0
Weather -> i_2
Job  -> i_3
Cities -> i_4
Vehicles -> i_5

I also manually searched the phrase list keywords in the sub and dispatch apps and there is none missing.
In short, the Dispatch merge logic performs as it should be.
There is no bug.

If @NC2408 still have an issue, please share with me the LUIS apps and I can take a look. This bug will be closed if there is no further issues.

**Comment by NC2408** [comment]
Hi,

I'll recreate our Dispatch model from our LUIS sources and test to see if any of our missing keywords appear in any of the lists in the way you describe and I'll let you know the outcome

**Comment by congysu** [comment]
@NC2408 does the solution work for you?  let us know.  Thanks.

**Comment by compulim** [comment]
@NC2408 we are closing this due to inactivity. If you are still having similar issues, please open a new issue and reference to this one.
